js/private/js_helpers.bzl

```diff
@@ -138,6 +138,38 @@ def gather_npm_package_store_deps(targets):
 
     return depset([], transitive = npm_package_store_deps)
 
+def copy_js_file_to_bin_action(ctx, file):
+    if ctx.label.workspace_name != file.owner.workspace_name or ctx.label.package != file.owner.package:
+        msg = """
+
+Expected to find source file {file_basename} in {this_package}, but instead it is in {file_package}.
+
+All source files in rules_js rules must be in the same package as the target.
```

Comment:
This sentence may be more confusing than helpful for users since "source file" is ambiguous and overloaded and means something else outside of Bazel.

@alexeagle WDYT the best error message is here? You're usually good with these.

The other thing to mention in this error message is the option to add the file to `no_copy_to_bin` to prevent copying to bin... and perhaps link to our docsite for the context: https://docs.aspect.build/rules/aspect_rules_js/docs/#javascript

Reply:
Guess you need to update the title of this PR as well :)

Reply:
"All source and data files that are not in the Bazel output tree must be in the same package as the target so that they can be copied to the output tree in an action" ?

Reply:
The docs link is in the next sentence.

I've updated this sentence and added the one at the end mentioning `no_copy_to_bin`.